### PR TITLE
Adds logging module in

### PR DESCRIPTION
**Pull request by bseeger**
Adds logging to verify.py.   

https://jira.duraspace.org/browse/FCREPO-2284

I'm coming to the conclusion that the logging module in python leaves something left to be desired (at least for what I'm trying to use it for).  However, this is an attempt at adding in logging and might therefore complete the above JIRA ticket.

### CODE REVIEW

**Comment by awoods** [comment]
Nice work, @bseeger . This may be good as is, or as @jwestgard mentions, it may make sense to segregate the logging and verbose capabilities. At this point, I could go either way... but we should probably have a brief conversation to discuss the eventual desired functionality and expectations around logging vs. user-verbosity. 

As a side note, there are quite a few whitespace changes in this PR that make the functional changes proportionally challenging to tease out.

**Comment by bseeger** [comment]
Yeah, the problem I was trying to solve was how to allow the user to get _some_ INFO messages and then even more detailed INFO messages, hence pulling in the verbose tag.  In my original version of this tool, I used the WARN level to display the individual resource information, which isn't really the right way to do things either. Perhaps a better option would be to have a separate flag for displaying detailed record info (not the verbose tag)?  I'll be available for a chat from Tuesday on, if you want to touch base about this.

File: verify.py [hunk 1]
```diff
@@ -354,37 +375,41 @@ def credentials(user):
...
                             'verification': verification
                             }
                     writer.writerow(row)
-                    
+
                 counter += 1
-    
+
     # Clear the resource counter display
-    print('')
```
Comment (jwestgard) [inline verify.py]: Hi Bethany,  this line needs to remain a print('') in order for the counter not to be overwritten when the script finishes.  The desired behavior is that the final count should remain visible when it exits.

Comment (bseeger) [inline verify.py]: Okay, I'll fix that. I never let it fully run so I didn't catch that (was using the LUMB dataset, which is huge).

**Comment by jwestgard** [comment]
I know that @bseeger is planning to implement changes to this PR, but FWIW I have tested the pull request and found everything to work well (with just the minor issues as noted above).

File: verify.py [hunk 2]
```diff
@@ -79,14 +79,15 @@ def get_directory_contents(localpath):
 #============================================================================
 
 class Config():
-    '''Object representing the options from import/export config and 
+    '''Object representing the options from import/export config and
        command-line arguments.'''
     def __init__(self, configfile, auth):
-        print("Loading configuration options from import/export config file...")
+        print('''Loading configuration options from import/export config file
```
Comment (jwestgard) [inline verify.py]: Python will print the line break inside the docstring, so the formatting of this line gets messed up.

An alternative is to concatenate long strings implicitly:

```
print("Loading configuration options from import/export config file",
        "'{0}'...".format(configfile))
```

I hope these comments are helpful and not annoying.

**Comment by bseeger** [comment]
Josh, I just tested this and am not seeing that behavior. If I specify another file name, it appears to work just fine.  Can you test again with my new commit?

**Comment by bseeger** [comment]
Changes in this commit: 
1) always creates output file - default name of 'verify_output.txt'
2) `-v` flag will display each record's info to stdout
3) errors and mismatches will go to both the log file and stdout.  If there's a mismatch, that records info will also go into the log file. 
4) made changes / fixes requested above.

**Comment by jwestgard** [comment]
@bseeger: I think I know what's going on.  The default logging level is WARN.  If you don't have any issues in the verification run, and you use the default, you'll get an empty log.  I wonder if there's a way to logically add some sort of indication that the verifciation ran without errors in such a case. Regardless, that can be an improvement for a different PR, if we decide that's useful.

**Comment by bseeger** [comment]
Josh, 
I think we can change the default level to INFO now - that's what I was testing with. It was only WARN in the event of not wanting to see all the resource info, which has now changed. I'll add a new commit with INFO as the default level.

**Comment by whikloj** [comment]
Might I suggest that you could use two different prefixes for the different messages in your logger. Then you could have then both at the INFO level, but only one handler would be set to display INFO level messages. That might allow you to keep the messages at the same level but separated.

**Comment by jwestgard** [comment]
Thanks, @whikloj, that is a good suggestion for the future, but I think since we've already worked out this issue another way (by separating out the other layer of INFO logging into a separate CSV output that can serve as a more permanent record of the verification run, and also by enabling a verbose option to get more feedback to screen), we'll go ahead as is for now.

**Comment by jwestgard** [comment]
I have finished testing and would be ready to merge this PR. @bseeger, could you confirm that lines 400-408 in verify.py should NOT be written to the log unless there is a problem, or whether the info-level comments should be written no matter what?  Once we sort that out, I would then merge the PR.

**Comment by bseeger** [comment]
Why do I see this in email but not here:  
from @jwestgard:  
--------
In verify.py:

> @@ -388,12 +407,6 @@ def credentials(user):
                     logger.info("  Verifying original to copy... {0} -- {1}".format(
                          verified, verification))
@bseeger , should these lines be written to the log when the resource is correctly verified? Right now they are included in the "if not verified" conditional and so only get written when there's a problem. I think we want them written at the INFO level when there is no mismatch, in which case we would just need an "else" in line 403.

------------------

My response -  I didn't think people wanted that info in the log. We also have the CSV file, so if they want that info, they can turn that feature on.  Thoughts?

**Comment by bseeger** [comment]
That being said, I don't have any specific opinion on whether or not we include that line in the log.  My original intention was that it would not be.

**Comment by jwestgard** [comment]
Thanks @bseeger.  I just wanted to confirm that was what you intended. I'll go ahead and merge this as is.  

My personal view is that the log and the csv have slightly different audiences.  For the csv maybe general stakeholders or repository managers who are interested in tracking the results of their migration, and for the log, people like us who need to debug the verification tool or developers who are testing import export and want to see everything that is happening.  But the content of the log is open to discussion and stakeholder feedback later, as you say.  For now let's get the basic functionality merged.

**Pull request merged**